### DISCUSSION TITLE
Clarification on ```continuous_index``` Handling Multi-Variable Inputs in CEBRA

### DISCUSSION THREAD

**Discussion by Barbod1380** [AdaptiveMotorControlLab/CEBRA#198] [Q&A] [answered]
Hi, I’m working with the CEBRA AI package, specifically in the multi-session training setup. I have a question regarding the ```continuous_index``` parameter.
From the documentation, I understand that ```continuous_index``` is an auxiliary variable (e.g., head direction or position). However, I am unsure how CEBRA handles multi-variable inputs for ```continuous_index```.

For example, if I pass a 2D numpy array with the shape (n, 3) as the ```continuous_index```, does the model:

- Treat the three columns as separate features?
- Or, consider it as a single feature with three components?

I would appreciate any clarification on how ```continuous_index``` is processed in this scenario.
Thank you for your help!

**Comment by stes** [comment, marked as answer]
Hi @Barbod1380 , by default, all continuous features are processed together, this is done due to how distance are calculated here https://github.com/AdaptiveMotorControlLab/CEBRA/blob/5f46c3257952a08dfa9f9e1b149a85f7f12c1053/cebra/distributions/index.py#L76-L99. 

It would be possible to specify a component-wise handling based on your application example. For that

- you would implement a distance matrix with the metric of your choice
- you would use this distance matrix in the continuous sampling

Please let me know if that addresses your question!
